iina/OpenSubSubtitle.swift

```diff
@@ -212,28 +220,29 @@ class OpenSubSupport {
 
       hash += fileSize
 
-      file.closeFile()
-
       resolver.fulfill(FileInfo(hashValue: String(format: "%016qx", hash), fileSize: fileSize))
     }
   }
 
-  func requestByName(_ fileURL: URL) -> Promise<[OpenSubSubtitle]> {
-    return requestIMDB(fileURL).then { imdb -> Promise<[OpenSubSubtitle]> in
+  func requestByName(_ fileURL: URL, _ playerCore: PlayerCore) -> Promise<[OpenSubSubtitle]> {
```

Comment:
Same here.

Reply:
From what I recall I believe I decided it would be ok to pass in PlayerCore because the parent class OnlineSubtitle was already interacting with PlayerCore. I'm used to dependency injection and strict layering where lower level code has no knowledge of higher level code, so easy to sell me on the idea that these low level methods should not be directly accessing PlayerCore. I will refactor all these methods to remove PlayerCore and make the caller pass in the required information.

Reply:
A lot of IINA code is a bit more tangled than it should be, unfortunately. Sorry for pushing the cleanup on you but as you can probably guess it's easier to fix these incrementally as people add new code, rather than having to come back and clean it up later.

Reply:
Not a problem asking me to do it. I am happy to do cleanup.  :-)